app/src/main/java/com/example/cocktails/ItemDetails/RecipeFragment.kt

```diff
@@ -116,7 +116,7 @@ class RecipeFragment : Fragment(), PreparationAdapter.ViewHolder.ClickListener {
 
     @RequiresApi(Build.VERSION_CODES.O)
     fun initViews(withTooltip: Boolean = true) {
-        if(cocktail.isReview){
+        if(cocktail.isReview||cocktail.isCustom){
```

Comment:
IMO, that means that you are reading the custom cocktail image from shared preference, using uri&bitmap, instead of using a url from firebase. if you are going to save all the images into a shared preference, application might crash after 2-3 cocktails, as we dont have enough space to hold images in our fat app :(

Reply:
I'm not done.